src/main/java/com/recipapp/service/AbstractDocumentService.java

```diff
@@ -0,0 +1,27 @@
+package com.recipapp.service;
+
+import com.recipapp.model.AbstractDocument;
+import com.recipapp.model.SequenceIdentifier;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+public abstract class AbstractDocumentService<O extends AbstractDocument, I> {
+
+    @Autowired
+    private SequenceService sequenceService;
+
+    public O save(O document) {
+        O modifiedDoc = this.preCreation(document);
```

Comment:
I don't understand what is the objective of preCreation method here.

Reply:
I thought I would need it, but then i didn't. It seems it's going to be handy someday